pkg/apis/core/validation/validation.go

```diff
@@ -6348,10 +6429,48 @@ func validateContainerResourceName(value core.ResourceName, fldPath *field.Path)
 	return allErrs
 }
 
+// validatePodResourceName verifies that:
+// 1. The resource name is a valid compute resource name for pod-level specification.
+// 2. The resource is supported by the PodLevelResources feature.
+func validatePodResourceName(value core.ResourceName, fldPath *field.Path) field.ErrorList {
+	allErrs := validateComputeResourceName(value, fldPath)
+	if len(allErrs) != 0 {
+		return allErrs
+	}
+
+	if !helper.IsNativeResource(value) {
+		return append(allErrs, field.Invalid(fldPath, value, "non-native resources are not supported for pod-level resources"))
+	}
+
+	resourceName, err := getResourceNameWithoutPrefix(value)
+	if err != nil {
+		return append(allErrs, field.Invalid(fldPath, value, err.Error()))
+	}
+
+	if !resourcehelper.IsSupportedPodLevelResource(v1.ResourceName(resourceName)) {
+		return append(allErrs, field.Invalid(fldPath, value, "must be a supported resource for pod-level resources"))
+	}
+
+	return allErrs
+}
+
+func getResourceNameWithoutPrefix(prefixedResource core.ResourceName) (string, error) {
+	segments := strings.Split(string(prefixedResource), "/")
+	if len(segments) > 2 {
+		return "", fmt.Errorf("must be a supported resource for pod-level resources")
```

Comment:
Is this function supposed to be general purpose, or only for validation pod resource? The function name suggests the former, but the error message the latter.